pkg/validations/base_test.go

```diff
@@ -55,6 +56,25 @@ func createEngine() (validationEngine, error) {
 	return ve, nil
 }
 
+func createEngineWithAllChecks() (validationEngine, error) {
```

Comment:
Can we combine this with previous function?  The differences seem like that can be passed in via function args.

Reply:
Not sure we need to keep the sync.Once in these tests.

Reply:
>Can we combine this with previous function? The differences seem like that can be passed in via function args.

I'll look at doing that.

>Not sure we need to keep the sync.Once in these tests.

Yeah probably not. It doesn't appear to be functionally doing anything, but I didn't touch it in case it was added for a reason I didn't understand. I'll go ahead and remove it.

Reply:
I removed use the sync package, and I refactored the engine construction functions to be more modular:

https://github.com/app-sre/deployment-validation-operator/blob/8fe3c3289499cb0c2ec8433920c0b7ccbb7e2551/pkg/validations/base_test.go#L27-L65